prov/rxm/src/rxm_domain.c

```diff
@@ -587,9 +594,20 @@ static int rxm_mr_regattr(struct fid *fid, const struct fi_mr_attr *attr,
 	ofi_mutex_init(&rxm_mr->amo_lock);
 	rxm_mr->iface = msg_attr.iface;
 	rxm_mr->device = msg_attr.device.reserved;
-	*mr = &rxm_mr->mr_fid;
-
-	if (rxm_domain->util_domain.info_domain_caps & FI_ATOMIC) {
+    *mr = &rxm_mr->mr_fid;
+
+    // get the GDR copy registration if possible, inspired from efa_mr_hmem_setup
+    if (flags & FI_HMEM_DEVICE_ONLY && cuda_is_gdrcopy_enabled()) {
+        int gdrerr =
+            cuda_gdrcopy_dev_register((struct fi_mr_attr *)attr, (uint64_t *)&rxm_mr->hmem_handle);
```

Comment:
Actually I think both EFA and RXM should consider adopting the new API in https://github.com/ofiwg/libfabric/commit/fed171974610ec010c2433afbb54ba6e713f9663

Reply:
Yes, I am actually opening a PR https://github.com/ofiwg/libfabric/pull/9456/files to update this function's signature. It's weird to use mr_attr as input

Reply:
Agree - let's wait for #9456 or rebase off those changes.